src/extension.ts

```diff
@@ -27,7 +27,7 @@ export function activate({ subscriptions }: vscode.ExtensionContext) {
 
   // counts the number of words in a passed in string
   const count = (text: string): number => {
-    return (text.match(/[a-zA-Z0-9]+/g) || []).length;
+    return (text.match(/[\w-]+(?<!-)\b/g) || []).length;
```

Comment:
You are right. `(?<!-)` and `\b` have the same effect if I want to exclude the starting `-` when they represent unordered lists in markdown.

Reply:
I just realized a problem. The regexp in my version will not work in French or other languages have different letters. 

The reason why I changed this line of code is that using space to count words failed in my use case to get an accurate enough result. 

I mix English with Chinese in writing, using Chinese characters rather than space as the boundary. 

e.g. `这是一个example that可能work`. 
There are 1 space but 3 words in English. 

Maybe the better way is to make the regexp configurable and use space as the default. I will fix it recently.

Reply:
Yes true, I also noticed that German words are split on "Umlauts" (i. e. Zürich counts as two words).

A configurable regexp would make sense.